cluster/discovery/discovery.go

```diff
@@ -0,0 +1,33 @@
+package discovery
+
+// NodeEntry is used to discovery nodes in the cluster
+type NodeEntry struct {
+	Id            string
+	Ip            string
+	GossipVersion string
```

Comment:
GossipVersion seems to be out of place here ? Also, isn't the version actually passed over wire ? (not sure why we need to put it here ..)